backend/apps/common/management/commands/dump_data.py

```diff
@@ -0,0 +1,158 @@
+        # Ensure output directory exists
+        output_path.parent.mkdir(parents=True, exist_ok=True)
+
+        temp_db = f"temp_{name}"
+        env = os.environ.copy()
+        if password:
+            env["PGPASSWORD"] = password
+
+        self.stdout.write(self.style.NOTICE(f"Creating temporary database: {temp_db}"))
+        try:
+            # 1) Create temp DB from template
+            self._psql(
+                host,
+                port,
+                user,
+                "postgres",
+                f"CREATE DATABASE {temp_db} TEMPLATE {name};",
+                env,
+            )
+
+            # 2) Hide emails
+            self.stdout.write(self.style.NOTICE("Hiding email fields in temp DB…"))
+            self._psql(host, port, user, temp_db, self._hide_emails(), env, via_stdin=True)
+
+            # 3) Dump selected tables
+            self.stdout.write(self.style.NOTICE(f"Creating dump at: {output_path}"))
+            dump_cmd = [
+                "pg_dump",
+                "-h",
+                host,
+                "-p",
+                port,
+                "-U",
+                user,
+                "-d",
+                temp_db,
+                "--compress=9",
+                "--clean",
+            ]
+            dump_cmd += [f"--table={table}" for table in tables]
+            dump_cmd += ["-f", str(output_path)]
+
+            run(dump_cmd, check=True, env=env)
+            self.stdout.write(self.style.SUCCESS(f"Dump created: {output_path}"))
+        except CalledProcessError as e:
+            message = f"Command failed: {e.cmd}"
+            raise CommandError(message) from e
+        finally:
+            # 4) Drop temp DB
+            self.stdout.write(self.style.NOTICE(f"Dropping temporary database: {temp_db}"))
+            try:
+                self._psql(
+                    host,
+                    port,
+                    user,
+                    "postgres",
+                    f"DROP DATABASE IF EXISTS {temp_db};",
+                    env,
+                )
+            except CalledProcessError:
+                # Best-effort cleanup
+                self.stderr.write(
+                    self.style.WARNING(f"Failed to drop temp DB {temp_db} (ignored).")
+                )
+
+    def _hide_emails(self) -> str:
+        # Uses a DO block to UPDATE every column named 'email' in non-system schemas
+        return """
+DO $$
+DECLARE
+    record RECORD;
+    statement TEXT;
+BEGIN
+    FOR record IN
+        SELECT quote_ident(n.nspname) AS schemaname,
+               quote_ident(c.relname) AS tablename,
+               quote_ident(a.attname) AS colname
+        FROM pg_attribute a
+        JOIN pg_class c ON c.oid = a.attrelid
+        JOIN pg_namespace n ON n.oid = c.relnamespace
+        WHERE a.attname = 'email'
+          AND a.attnum > 0
+          AND NOT a.attisdropped
+          AND n.nspname NOT IN ('pg_catalog','information_schema','pg_toast')
```

Comment:
These magic names and pretty much the entire PL/SQL spell doesn't fit our clean and readable code approach. Yes, it's probably performs the best but it's not the only goal we have.